File: DhiaAmine/piAmine/src/entities/Etablissement.java
package entities;

public class Etablissement {

    private int ide;
    private int id_prop;
    private int numero;
    private String type;
    private String nom;
    private String adresse;
    private String description;
    private String avis;

    public Etablissement(int id_prop, int numero, String type, String nom, String adresse, String description, String avis) {
        this.id_prop = id_prop;
        this.numero = numero;
        this.type = type;
        this.nom = nom;
        this.adresse = adresse;
        this.description = description;
        this.avis = avis;
    }

    public Etablissement(int ide) {
        this.ide = ide;
    }

    public Etablissement(int ide, int id_prop, int numero, String type, String nom, String adresse, String description, String avis) {
        this.ide = ide;
        this.id_prop = id_prop;
        this.numero = numero;
        this.type = type;
        this.nom = nom;
        this.adresse = adresse;
        this.description = description;
        this.avis = avis;
    }

    public int getIde() {
        return ide;
    }

    public void setIde(int ide) {
        this.ide = ide;
    }

    public int getId_prop() {
        return id_prop;
    }

    public void setId_prop(int id_prop) {
        this.id_prop = id_prop;
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAvis() {
        return avis;
    }

    public void setAvis(String avis) {
        this.avis = avis;
    }
   
    @Override
    public String toString() {
        return "Etablissement{" + "ide=" + ide + ", id_prop=" + id_prop + ", numero=" + numero + ", type=" + type + ", nom=" + nom + ", adresse=" + adresse + ", description=" + description + ", avis=" + avis + '}';
    }

   // public int getIde() {
    //    throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    //}

}

File: SymfonyDhia/SymDhia/src/Entity/Etablissement.php
<?php

namespace App\Entity;

use App\Repository\EtablissementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * @ORM\Entity(repositoryClass=EtablissementRepository::class)
 */
class Etablissement
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="integer")
     */
    private $id_prop;

    /**
     * @ORM\Column(type="integer")
     */
    private $numero;

    /**
     * @Assert\NotBlank(message=" type doit etre non vide")
     * @Assert\Length(
     *      min = 4,
     *      minMessage=" Entrer un titre au mini de 4 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $type;

    /**
     * @Assert\NotBlank(message="nom   :doit etre non vide")
     * @ORM\Column(type="string", length=1000)
     */
    private $nom;

    /**
     * @Assert\NotBlank(message=" adresse doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      minMessage=" Entrer une adresse au mini de 7 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $adresse;

    /**
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=100" )
     * @ORM\Column(type="string", length=1000)
     */
    private $description;

    /**
     * @Assert\NotBlank(message=" nom doit etre non vide")
     * @Assert\Length(
     *      min = 5,
     *      minMessage=" Entrer un nom au mini de 5 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $avis;

    /**
     * @ORM\OneToMany(targetEntity=Commande::class, mappedBy="etablissement",cascade={"all"},orphanRemoval=true)
     */
    private $commandes;

    public function __construct()
    {
        $this->commandes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getIdProp(): ?int
    {
        return $this->id_prop;
    }

    public function setIdProp(int $id_prop): self
    {
        $this->id_prop = $id_prop;

        return $this;
    }

    public function getNumero(): ?int
    {
        return $this->numero;
    }

    public function setNumero(int $numero): self
    {
        $this->numero = $numero;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getAvis(): ?string
    {
        return $this->avis;
    }

    public function setAvis(string $avis): self
    {
        $this->avis = $avis;

        return $this;
    }

    /**
     * @return Collection<int, Commande>
     */
    public function getCommandes(): Collection
    {
        return $this->commandes;
    }

    public function addCommande(Commande $commande): self
    {
        if (!$this->commandes->contains($commande)) {
            $this->commandes[] = $commande;
            $commande->setEtablissement($this);
        }

        return $this;
    }

    public function removeCommande(Commande $commande): self
    {
        if ($this->commandes->removeElement($commande)) {
            // set the owning side to null (unless already changed)
            if ($commande->getEtablissement() === $this) {
                $commande->setEtablissement(null);
            }
        }

        return $this;
    }
}
